lib/xdr/transactions/ed25519_contract_id.ex

```diff
@@ -0,0 +1,104 @@
+defmodule StellarBase.XDR.Ed25519ContractID do
+  @moduledoc """
+  Representation of Stellar `Ed25519ContractID` type.
+  """
+  alias StellarBase.XDR.{Hash, UInt256}
+
+  @behaviour XDR.Declaration
+
+  @struct_spec XDR.Struct.new(
+                 network_id: Hash,
+                 ed25519: UInt256,
+                 salt: UInt256
+               )
+
+  @type t :: %__MODULE__{
+          network_id: Hash.t(),
+          ed25519: UInt256.t(),
+          salt: UInt256.t()
+        }
+
+  defstruct [:network_id, :ed25519, :salt]
+
+  @spec new(
+          network_id :: Hash.t(),
+          ed25519 :: UInt256.t(),
+          salt :: UInt256.t()
+        ) :: t()
+  def new(
+        %Hash{} = network_id,
+        %UInt256{} = ed25519,
+        %UInt256{} = salt
+      ) do
+    %__MODULE__{
+      network_id: network_id,
+      ed25519: ed25519,
+      salt: salt
+    }
+  end
```

Comment:
@keliumJU Try to keep fewer lines of code here. Write everything on one line when possible and the linter allows it!
It applies also for `encode_xdr/1` and `encode_xdr!/1` functions on many modules.


Also, please use the `, do:` notation in the `new` function.





```elixir
@spec new(network_id :: Hash.t(), ed25519 :: UInt256.t(), salt :: UInt256.t()) :: t()
  def new(%Hash{} = network_id, %UInt256{} = ed25519, %UInt256{} = salt),
    do: %__MODULE__{
      network_id: network_id,
      ed25519: ed25519,
      salt: salt
    }
```